src/Components/SingleArticle.jsx

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import axios from "axios";
+import "./SingleArticle.css";
+
+export default function SingleArticle() {
+  const { article_id } = useParams();
+  const [article, setArticle] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    axios
+      .get(
+        `https://be-nc-news-candy-sep.onrender.com/api/articles/${article_id}`,
+      )
+      .then(({ data }) => {
+        setArticle(data.article);
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        setError("Article not found");
+        setIsLoading(false);
+      });
```

Comment:
great api call. It'd be even better if all the api calls could be abstracted into another file to keep the components neat

Reply:
Ok, will work on this, Thank you!